vnext/Shared/Modules/BlobModule.cpp

```diff
@@ -0,0 +1,454 @@
+         auto base64String = Common::Unicode::Utf16ToUtf8(std::move(winrtString));
+
+         wsProxy->SendBinary(std::move(base64String), socketID);
+       }},
+
+      {"createFromParts",
+       // As of React Native 0.67, instance is set AFTER CxxModule::getMethods() is invoked.
+       // Use getInstance() directly once
+       // https://github.com/facebook/react-native/commit/1d45b20b6c6ba66df0485cdb9be36463d96cf182 becomes available.
+       [persistor = m_blobPersistor, weakState = weak_ptr<SharedState>(m_sharedState)](dynamic args) {
+         auto parts = jsArgAsArray(args, 0); // Array<Object>
+         auto blobId = jsArgAsString(args, 1);
+         vector<uint8_t> buffer{};
+
+         for (const auto &part : parts) {
+           auto type = part[typeKey].asString();
+           if (blobKey == type) {
+             auto blob = part[dataKey];
+             winrt::array_view<uint8_t> bufferPart;
+             try {
+               bufferPart = persistor->ResolveMessage(
+                   blob[blobIdKey].asString(), blob[offsetKey].asInt(), blob[sizeKey].asInt());
+             } catch (const std::invalid_argument &e) {
+               if (auto sharedState = weakState.lock()) {
+                 Modules::SendEvent(sharedState->Module->getInstance(), "blobFailed", e.what());
+               }
+               return;
+             }
+
+             buffer.reserve(buffer.size() + bufferPart.size());
+             buffer.insert(buffer.end(), bufferPart.begin(), bufferPart.end());
+           } else if ("string" == type) {
+             auto data = part[dataKey].asString();
+
+             buffer.reserve(buffer.size() + data.size());
+             buffer.insert(buffer.end(), data.begin(), data.end());
+           } else {
+             if (auto state = weakState.lock()) {
+               auto message = "Invalid type for blob: " + type;
+               Modules::SendEvent(state->Module->getInstance(), "blobFailed", std::move(message));
+             }
+             return;
+           }
+         }
+
+         persistor->StoreMessage(std::move(buffer), std::move(blobId));
+       }},
+
+      {"release",
+       [persistor = m_blobPersistor](dynamic args) // blobId: string
+       {
+         auto blobId = jsArgAsString(args, 0);
+
+         persistor->RemoveMessage(std::move(blobId));
+       }}};
+}
+
+#pragma endregion CxxModule
+
+#pragma endregion BlobModule
+
+#pragma region MemoryBlobPersistor
+
+#pragma region IBlobPersistor
+
+winrt::array_view<uint8_t> MemoryBlobPersistor::ResolveMessage(string &&blobId, int64_t offset, int64_t size) {
+  if (offset < 0 || size < 1)
+    return {};
+
+  scoped_lock lock{m_mutex};
+
+  auto dataItr = m_blobs.find(std::move(blobId));
+  if (dataItr != m_blobs.cend()) {
+    auto &bytes = (*dataItr).second;
+
+    auto endBound = static_cast<size_t>(offset + size);
+    // Out of bounds.
+    if (endBound > bytes.size())
+      return {};
```

Comment:
>return {}; [](http://example.com/codeflow?start=6&length=10)

Should we just return till the end here ?

Reply:
Sounds reasonable.